lib/ansible/modules/network/nxos/nxos_acl.py

```diff
@@ -458,13 +459,35 @@ def main():
     delta_options = {}
 
     if not existing_core.get('remark'):
-        delta_core = dict(
+        dcore = dict(
             set(proposed_core.items()).difference(
                 existing_core.items())
         )
-        delta_options = dict(
-            set(proposed_options.items()).difference(
-                existing_options.items())
+        if not dcore:
+            # check the diff in the other way just in case
+            dcore = dict(
+                set(existing_core.items()).difference(
```

Comment:
Wondering why check the other way around? Is this for idempotence?

Reply:
It is not just for idempotence. There are 2 ways the existing acl can be changed.
1. we can add 1 or more parameters to the existing acl
2. we can remove 1 or more parameters from the existing acl

The current code is only taking care of the first change. The 2nd one is being ignored and so if user wants to remove some parameters, there is no way, the playbook run does not even throw an error, it does not change anything and silently exits. The problem exists for both core and options.

Here is an example:
```
>>> proposed_options = {'c' : 'd'}

>>> existing_options = {'a' : 'b', 'c' : 'd'}

>>> dict(set(proposed_options.items()).difference(existing_options.items()))
{}

>>> dict(set(existing_options.items()).difference(proposed_options.items()))
{'a': 'b'}
```

So as you can see, the 2nd statement detects if the difference is in the removing part. 
Since this is intended behavior, this is a bug. 
You can check the documentation:
https://github.com/ansible/ansible/blob/cdd21e2170b9a47d84624c8c2f792a66f2b0fe57/lib/ansible/modules/network/nxos/nxos_acl.py#L44-L45

Reply:
Thanks. Makes sense.